packages/web3-eth-accounts/src/index.js

```diff
@@ -241,21 +254,41 @@ Accounts.prototype.signTransaction = function signTransaction(tx, privateKey, ca
 
 
     // Resolve immediately if nonce, chainId, price and signing options are provided
-    if (tx.nonce !== undefined && tx.chainId !== undefined && tx.gasPrice !== undefined && hasTxSigningOptions) {
+    if (
+        tx.nonce !== undefined &&
+        tx.chainId !== undefined &&
+        (
+            tx.gasPrice !== undefined ||
+            (
+                tx.maxFeePerGas !== undefined &&
+                tx.maxPriorityFeePerGas !== undefined
+            )
+        ) &&
+        hasTxSigningOptions
+    ) {
         return Promise.resolve(signed(tx));
     }
 
     // Otherwise, get the missing info from the Ethereum Node
     return Promise.all([
         isNot(tx.chainId) ? _this._ethereumCall.getChainId() : tx.chainId,
-        isNot(tx.gasPrice) ? _this._ethereumCall.getGasPrice() : tx.gasPrice,
+        // isNot(tx.gasPrice) ? _this._ethereumCall.getGasPrice() : tx.gasPrice,
+        _handleTxPricing(tx),
         isNot(tx.nonce) ? _this._ethereumCall.getTransactionCount(_this.privateKeyToAccount(privateKey).address) : tx.nonce,
         isNot(hasTxSigningOptions) ? _this._ethereumCall.getNetworkId() : 1
     ]).then(function(args) {
         if (isNot(args[0]) || isNot(args[1]) || isNot(args[2]) || isNot(args[3])) {
             throw new Error('One of the values "chainId", "networkId", "gasPrice", or "nonce" couldn\'t be fetched: ' + JSON.stringify(args));
         }
-        return signed(_.extend(tx, {chainId: args[0], gasPrice: args[1], nonce: args[2], networkId: args[3]}));
+        return signed(_.extend(
+            tx,
+            {
+                chainId: args[0],
+                nonce: args[2],
+                networkId: args[3],
+                ...args[1] // Will either be gasPrice or maxFeePerGas and maxPriorityFeePerGas
```

Comment:
ocd killing me on this, can you push `_handleTxPricing` to the end of `Promise.all(..)` so it inherits the 3rd index? Or just re-order this